packages/astro/src/core/build/static-build.ts

```diff
@@ -172,8 +177,10 @@ async function ssrBuild(
 					assetFileNames: `${settings.config.build.assets}/[name].[hash][extname]`,
 					...viteConfig.build?.rollupOptions?.output,
 					entryFileNames(chunkInfo) {
-						if (chunkInfo.facadeModuleId === resolvedPagesVirtualModuleId) {
-							return opts.buildConfig.serverEntry;
+						if (chunkInfo.facadeModuleId?.startsWith(ASTRO_PAGE_RESOLVED_MODULE_ID)) {
+							return makeAstroPageEntryPointFileName(chunkInfo.facadeModuleId);
+						} else if (chunkInfo.facadeModuleId === SSR_VIRTUAL_MODULE_ID) {
+							return opts.settings.config.build.serverEntry;
 						} else if (chunkInfo.facadeModuleId === RESOLVED_MIDDLEWARE_MODULE_ID) {
 							return 'middleware.mjs';
 						} else if (chunkInfo.facadeModuleId === RESOLVED_RENDERERS_MODULE_ID) {
```

Comment:
With the `plugin-middleware` refactor, this condition wont pass anymore. It'll still pass if you have a `src/middleware.ts` because the else condition here catches it. But if I have `src/middleware/index.ts`, it won't output `middleware.mjs` anymore (`index.mjs` instead) (tested in `middleware` example). Maybe it's simpler to keep the `plugin-middleware` changes to handle this for now.

Reply:
Good catch! Completely missed that

Reply:
@ematipico I might've pointed the wrong thing, but with the new commit, this scenario still happens:

> But if I have `src/middleware/index.ts`, it won't output `middleware.mjs` anymore (`index.mjs` instead) (tested in `middleware` example).

In the `middleware` example with `src/middleware/index.ts`, the middleware is output as `dist/server/index.mjs`, which I don't think is intentional, and we want `dist/server/middlewares.mjs` instead?

I was suggesting keeping the previous `plugin-middleware` code, or maybe there's a property in `chunkInfo` that we can use to identify is a middleware?

Reply:
Maybe:

```js
if (chunkInfo.facadeModuleId?.includes('middleware') && chunkInfo.exports.includes('onRequest')) {
  return 'middleware.mjs'
}
```

A bit fragile though.